test/api/clustering_test.cpp

```diff
@@ -1,19 +1,319 @@
+    resulting_clusters = simple_clustering_method(junctions);
 
     std::vector<Cluster> expected_clusters{};
     // TODO (irallia): We need to implement operator== for Clusters.
     // EXPECT_EQ(expected_clusters, resulting_clusters);
 }
+
+std::string const chrom1 = "chr1";
+int32_t const chrom1_position1 = 12323443;
+int32_t const chrom1_position2 = 94734377;
+int32_t const chrom1_position3 = 112323345;
+std::string const chrom2 = "chr2";
+int32_t const chrom2_position1 = 234432;
+std::string const read_name_1 = "m2257/8161/CCS";
+std::string const read_name_2 = "m41327/11677/CCS";
+std::string const read_name_3 = "m21263/13017/CCS";
+std::string const read_name_4 = "m38637/7161/CCS";
+std::string const read_name_5 = "m23412/9534/CCS";
+std::string const read_name_6 = "m1245/5634/CCS";
+std::string const read_name_7 = "m8765/9765/CCS";
+std::string const read_name_8 = "m13456/11102/CCS";
+
+std::vector<Junction> prepare_input_junctions()
+{
+    std::vector<Junction> input_junctions
+    {   Junction{Breakend{chrom1, chrom1_position1 - 5, strand::forward},
+                    Breakend{chrom2, chrom2_position1 + 8, strand::forward}, ""_dna5, read_name_1}, //cluster 1
+        Junction{Breakend{chrom1, chrom1_position1 + 2, strand::forward},
+                    Breakend{chrom2, chrom2_position1 - 3, strand::forward}, ""_dna5, read_name_2}, //cluster 1
+        Junction{Breakend{chrom1, chrom1_position1 + 9, strand::forward},
+                    Breakend{chrom2, chrom2_position1 + 1, strand::forward}, ""_dna5, read_name_3}, //cluster 1
+        Junction{Breakend{chrom1, chrom1_position1 + 5, strand::forward},
+                    Breakend{chrom2, chrom2_position1 - 1, strand::reverse}, ""_dna5, read_name_4}, //cluster 2
+        Junction{Breakend{chrom1, chrom1_position1 + 92, strand::forward},
+                    Breakend{chrom2, chrom2_position1 + 3, strand::forward}, ""_dna5, read_name_5}, //cluster 3
+        Junction{Breakend{chrom1, chrom1_position2 - 2, strand::forward},
+                    Breakend{chrom2, chrom1_position3 + 8, strand::reverse}, ""_dna5, read_name_6}, //cluster 4
+        Junction{Breakend{chrom1, chrom1_position2 + 3, strand::forward},
+                    Breakend{chrom2, chrom1_position3 - 1, strand::reverse}, ""_dna5, read_name_7}, //cluster 4
+        Junction{Breakend{chrom1, chrom1_position2 + 6, strand::forward},
+                    Breakend{chrom2, chrom1_position3 + 2, strand::reverse}, ""_dna5, read_name_8} //cluster 4
+    };
+
+    std::sort(input_junctions.begin(), input_junctions.end());
+    return input_junctions;
+}
+
+
+TEST(clustering, partitioning)
+{
+    std::vector<Junction> input_junctions = prepare_input_junctions();
+    std::vector<std::vector<Junction>> partitions = partition_junctions(input_junctions);
+
+    std::vector<std::vector<Junction>> expected_partitions
+    {
+        {   Junction{Breakend{chrom1, chrom1_position1 - 5, strand::forward},
```

Comment:
Since we are following seqan3 format claims:
```suggestion
        {   
            Junction{Breakend{chrom1, chrom1_position1 - 5, strand::forward},
```